### PR TITLE
Bug/menu item missing aria labels

### PR DESCRIPTION
**Pull request by blunteshwar**
## Description
Added aria-haspopup,aria-expanded and aria-controls to MenuItem with subitem.

## Related issue(s)
[#3248](https://github.com/adobe/spectrum-web-components/issues/3248)

-

## Motivation and context
The following attributes { aria-expanded , aria-haspopup and aria-controls } did not exist for a MenuItem with subitem so I added them.

## How has this been tested?

Uploading Screen Recording 2023-07-12 at 8.18.21 PM.mov…

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Screenshots (if appropriate)
Screenshot 2023-07-07 at 2 34 19 PM

## Types of changes

-   [ x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [ x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [ x] My code follows the code style of this project.
-   [ x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [ x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ x] All new and existing tests passed.
-   [ x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: packages/accordion/src/AccordionItem.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2020 Adobe. All rights reserved.
+Copyright. 2020 Adobe. All rights reserved.
```
Comment (najikahalsema) [inline packages/accordion/src/AccordionItem.ts:2]: Can you remove this? Seems like a typo

**Review by najikahalsema — CHANGES_REQUESTED** [review]
How did you test this? Did you utilise the accessibility tree or screen reader?

File: packages/menu/src/MenuItem.ts [hunk 2]
```diff
@@ -425,7 +430,10 @@ export class MenuItem extends LikeAnchor(Focusable) {
             this.handleSubmenuPointerenter
         );
         submenu.addEventListener('change', this.handleSubmenuChange);
+        submenu.setAttribute('id', `${this.id}-submenu`);
```
Comment (Westbrook) [inline packages/menu/src/MenuItem.ts:433]: This will cause unintended side effects if the consumer has already applied an ID to this element.

**Review by Westbrook — COMMENTED** [review]
There’s a bunch of white space addition, was the intentional?

File: packages/menu/src/MenuItem.ts (hunk 2 above)
Comment (Rajdeepc) [reply packages/menu/src/MenuItem.ts:433]: @blunteshwar Please check for ID property before applying this attribute

File: packages/menu/src/MenuItem.ts [hunk 3]
```diff
@@ -330,6 +330,9 @@ export class MenuItem extends LikeAnchor(Focusable) {
             flatten: true,
         });
         this.hasSubmenu = this.open || !!assignedElements.length;
```
Comment (Rajdeepc) [inline packages/menu/src/MenuItem.ts:332]: Do a null check on assignedElements first otherwise this can throw a null pointer exception

**Review by Rajdeepc — CHANGES_REQUESTED** [review]

File: packages/accordion/src/AccordionItem.ts (hunk 1 above)
Comment (blunteshwar) [reply packages/accordion/src/AccordionItem.ts]: done

**Comment by blunteshwar** [comment]
> 



> There’s a bunch of white space addition, was the intentional?

extra white spaces have been removed now

**Comment by blunteshwar** [comment]
> How did you test this? Did you utilise the accessibility tree or screen reader?

I have used both screen reader and accessibility tree

File: packages/menu/src/MenuItem.ts [hunk 4]
```diff
@@ -425,6 +430,11 @@ export class MenuItem extends LikeAnchor(Focusable) {
             this.handleSubmenuPointerenter
         );
         submenu.addEventListener('change', this.handleSubmenuChange);
+        if (!submenu.id) {
+            // if the consumer has already applied an ID to this element then we won't explicitly aplly ID
```
Comment (Rajdeepc) [inline packages/menu/src/MenuItem.ts:434]: please remove the comments here!

File: packages/menu/src/Menu.ts [hunk 5]
```diff
@@ -631,6 +631,7 @@ export class Menu extends SpectrumElement {
         }
         item.focused = this.hasVisibleFocusInTree();
         this.setAttribute('aria-activedescendant', item.id);
+
```
Comment (Rajdeepc) [inline packages/menu/src/Menu.ts:634]: remove this white space

**Review by Rajdeepc — CHANGES_REQUESTED** [review]

**Review by Rajdeepc — APPROVED** [review]

**Review by Rajdeepc — APPROVED** [review]

**Review by najikahalsema — APPROVED** [review]

**Pull request merged**